### PR TITLE
Add Sync grant to the token generator

### PR DESCRIPTION
**Pull request by mamont**

### CODE REVIEW

File: src/Twilio.Auth/SyncGrant.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        }
+
+        public object GetPayload()
+        {
+            var payload = new Dictionary<string, string>();
+            if (this.ServiceSid != null)
+            {
+                payload.Add("service_sid", this.ServiceSid);
+            }
+            if (this.ServiceSid != null)
```
Comment (dprothero) [inline src/Twilio.Auth/SyncGrant.cs]: Shouldn't this be: `if (this.EndpointId != null)`

File: src/Twilio.Auth/SyncGrant.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            {
+                payload.Add("endpoint_id", this.EndpointId);
+            }
+            if (this.ServiceSid != null)
```
Comment (dprothero) [inline src/Twilio.Auth/SyncGrant.cs]: `if (this.DeploymentRoleSid != null)`

File: src/Twilio.Auth/SyncGrant.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            {
+                payload.Add("deployment_role_sid", this.DeploymentRoleSid);
+            }
+            if (this.ServiceSid != null)
```
Comment (dprothero) [inline src/Twilio.Auth/SyncGrant.cs]: `if (this.PushCredentialSid != null)`

File: src/Twilio.Auth/SyncGrant.cs (hunk 1 above)
Comment (dprothero) [inline src/Twilio.Auth/SyncGrant.cs]: I checked IpMessagingGrant.cs and it's the same there. Weird.

Comment (ajtack) [inline src/Twilio.Auth/SyncGrant.cs]: Indeed. Strange way to write such code. But I see what it's trying to do.

**Pull request closed without merging**